pyomo/contrib/solver/config.py

```diff
@@ -31,21 +31,45 @@
 
 
 def TextIO_or_Logger(val):
-    ans = []
-    if not isinstance(val, Sequence):
+    """
+    Validates and converts input into a list of valid output streams.
+
+    Accepts:
+      - sys.stdout
+      - Instances of io.TextIOBase
+      - logging.Logger (wrapped as LogStream)
+      - Boolean values (`True` → sys.stdout)
+
+    Returns:
+      - A list of validated output streams.
+
+    Raises:
+      - ValueError if an invalid type is provided.
+    """
+    if isinstance(val, (sys.stdout.__class__, io.TextIOBase, logging.Logger)):
         val = [val]
+
+    elif isinstance(val, Sequence) and not isinstance(val, (str, bytes)):
+        val = list(val)
+
+    else:
+        val = [val]
```

Comment:
This seems redundant and messy:
- we are checking types here, and again in the loop
- why are we forcing a copy of Sequences?

Why is this not just
```python
if not isinstance(val, Sequence) or isinstance(val, (str, bytes)):
    val = [val]
```